src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -4701,6 +4724,12 @@ CachedRelationNamespaceLookupExtended(const char *relationName, Oid relnamespace
 {
 	/* force callbacks to be registered, so we always get notified upon changes */
 	InitializeCaches();
+	Oid citusExtensionOid = get_extension_oid("citus", true);
```

Comment:
Following the workflow from when we start initializing tables from MetadataCacheData, they always require a call to `CachedRelationLookup()` which calls CachedRelationNamespaceLookup() which then calls CachedRelationNamespaceLookupExtended(). It seemed safer to place it in a function that's related to the transaction Ids we were concerned about.  I chose the Extended function because I noticed that other functions not relating to `MetadataCacheData` would call `CachedRelationLookup` and `CachedRelationNamespaceLookup`. I took a look at src/backend/distributed/commands/extension.c but I couldn't find an existing function for during `Create Extension` only for after.

Reply:
You can place it as one of the steps in: https://github.com/citusdata/citus/blob/master/src/backend/distributed/metadata/metadata_cache.c#L1906 ?